packages/vite/src/hbs.ts

```diff
@@ -99,6 +98,24 @@ async function maybeSynthesizeComponentJS(context: PluginContext, source: string
   if (!templateResolution) {
     return null;
   }
+
+  const resolvedId = templateResolution.id.split('?')[0];
+  templateResolution.id = resolvedId;
+
+  const pkg = resolverLoader.resolver.packageCache.ownerOfFile(resolvedId);
+  const isInComponents = pkg?.isV2App() && resolvedId.slice(pkg?.root.length).startsWith('/components');
```

Comment:
this will always be false for addons, which is good, because they should not do template colocation